src/Domain/Stack/LiipTransformerStackExecutor.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Liip\ImagineBundle\Domain\Stack;
+
+use Imagine\Image\ImageInterface;
+use Imagine\Image\ImagineInterface;
+use Liip\ImagineBundle\Binary\MimeTypeGuesserInterface;
+use Liip\ImagineBundle\Domain\ImageReference;
+use Liip\ImagineBundle\Domain\ImageReferenceFile;
+use Liip\ImagineBundle\Domain\Imagine\Filter\FilterExecutor;
+use Liip\ImagineBundle\Domain\PostProcessor\PostProcessor;
+use Liip\ImagineBundle\Imagine\Filter\FilterConfiguration;
+
+class LiipTransformerStackExecutor implements TransformerStackExecutor
+{
+    /**
+     * @var array<string, FilterExecutor>
+     */
+    private array $filters = [];
+
+    /**
+     * @var array<string, PostProcessor>
+     */
+    private array $postProcessors = [];
+
+    public function __construct(
+        private FilterConfiguration $filterConfiguration,
+        private ImagineInterface $imagine,
+        private MimeTypeGuesserInterface $mimeTypeGuesser,
+    ) {}
+
+    /**
+     * Adds a loader to handle the given filter.
+     */
+    public function addFilter(string $filterName, FilterExecutor $executor): void
+    {
+        $this->filters[$filterName] = $executor;
+    }
+
+    /**
+     * Adds a post-processor to handle binaries.
+     */
+    public function addPostProcessor(string $name, PostProcessor $postProcessor): void
+    {
+        $this->postProcessors[$name] = $postProcessor;
+    }
+
+    /**
+     * Apply the stack to the image.
+     */
+    public function apply(string $stackName, ImageReference $sourceImageReference): ImageReference
+    {
+        $config = $this->filterConfiguration->get($stackName);
+
+        $config += [
+            'quality' => 100,
+            'animated' => false,
+        ];
+
+        if ($sourceImageReference instanceof ImageReferenceFile) {
+            $image = $this->imagine->open($sourceImageReference->getPath());
+        } else {
+            $image = $this->imagine->load($sourceImageReference->getContent());
+        }
+
+        $image = $this->applyFilters($image, $config);
+        $resultImageReference = $this->exportConfiguredImageBinary($sourceImageReference, $image, $config);
```

Comment:
we need this step before we apply post processors. and it also depends on the stack configuration. so i think we need all 3 things in the same place and can not separate them.